src/select.coffee

```diff
@@ -140,16 +153,39 @@ Ember.AddeparMixins.ResizeHandlerMixin, Ember.Widgets.KeyboardHelper,
       @get('originalItemViewClass')
   .property 'showTooltip'
 
+  ###*
+  * Function to be called when we want to open the dropdown, i.e. we have to
+  * clean up the query, render the dropdown...
+  * @function
+  ###
+  openDropdown: ->
+    @set 'showDropdown', yes
+    # Clear the query content when users open the dropdown so that users can
+    # start typing their new query right away.
+    @set 'query', ''
+
+    @updateDropdownLayout()
+
+    # Set the focus to the search field so that users can start typing their
+    # query right away.
+    @_setSearchFieldFocus = Ember.run.schedule 'afterRender', this, ->
+      if (@get('_state') or @get('state')) is 'inDOM'
+        @$('.ember-select-search .ember-text-field:eq(0)').focus()
+
   # This doesn't clean correctly if `optionLabelPath` changes
   willDestroy: ->
     propertyName = 'contentProxy'
     if @cacheFor propertyName
       contentProxy = @get propertyName
       contentProxy.destroy()
+
+    # cancel the set focus schedule if we have it set up in openDropdown
+    Ember.run.cancel @_setSearchFieldFocus if @_setSearchFieldFocus
+
     @_super()
 
-  updateDropdownLayout: Ember.observer ->
-    return if (@get('_state') or @get('state')) isnt 'inDOM' or @get('showDropdown') is no
+  updateDropdownLayout: ->
+    return if (@get('_state') or @get('state')) isnt 'inDOM'
```

Comment:
Can we make `(@get('_state') or @get('state')) isnt 'inDOM'` into a computed property and use it throughout this file?